BlueprintUI/Sources/Layout/Flow.swift

```diff
@@ -0,0 +1,390 @@
+
+                    row = rowLayout(
+                        origin: .init(x: 0, y: totalHeight),
+                        maxWidth: constraint.maximum.width
+                    )
+                }
+
+                row.addItem(of: elementSize)
+            }
+
+            return frames + row.itemFrames()
+        }
+
+        private func size(
+            for elements: [ElementSize],
+            in constraint: SizeConstraint
+        ) -> CGSize {
+            frames(
+                for: elements,
+                in: constraint
+            ).reduce(.zero) { reduced, current in
+                CGSize(
+                    width: max(current.maxX, reduced.width),
+                    height: max(reduced.height, current.maxY)
+                )
+            }
+        }
+
+        // MARK: Legacy Layout
+
+        func measure(
+            in constraint: SizeConstraint,
+            items: [(
+                traits: Traits,
+                content: Measurable
+            )]
+        ) -> CGSize {
+            size(
+                for: items.map { $0.content.measure(in:) },
+                in: constraint
+            )
+        }
+
+        func layout(
+            size: CGSize,
+            items: [(
+                traits: Traits,
+                content: Measurable
+            )]
+        ) -> [LayoutAttributes] {
+            frames(
+                for: items.map { $0.content.measure(in:) },
+                in: .init(size)
+            ).map(LayoutAttributes.init(frame:))
+        }
+
+        // MARK: Private Methods
+
+        private func rowLayout(
+            origin: CGPoint,
+            maxWidth: CGFloat
+        ) -> RowLayout {
+            .init(
+                origin: origin,
+                maxWidth: maxWidth,
+                itemSpacing: horizontalSpacing,
+                horizontalAlignment: horizontalAlignment,
+                rowAlignment: rowAlignment
+            )
+        }
+    }
+}
+
+// MARK: - RowLayout
+
+extension Flow.Layout {
+
+    /// Helper for computing the frames for a row of items.
+    fileprivate struct RowLayout {
+
+        let origin: CGPoint
+        let maxWidth: CGFloat
+        let itemSpacing: CGFloat
+        let horizontalAlignment: Flow.HorizontalAlignment
+        let rowAlignment: Flow.RowAlignment
+
+        init(
+            origin: CGPoint,
+            maxWidth: CGFloat,
+            itemSpacing: CGFloat,
+            horizontalAlignment: Flow.HorizontalAlignment,
+            rowAlignment: Flow.RowAlignment
+        ) {
+            self.origin = origin
+            self.maxWidth = maxWidth
+            self.itemSpacing = itemSpacing
+            self.horizontalAlignment = horizontalAlignment
+            self.rowAlignment = rowAlignment
+        }
+
+        private(set) var height: CGFloat = 0
+        private(set) var items: [Item] = []
+        private var totalItemWidth: CGFloat = 0
+
+        private var itemCount: CGFloat {
```

Comment:
Nit: Feels iffy to have a count be a float. Can we instead directly pull `items.count` and convert to `CGFloat` as needed?

Reply:
Sure!

- [ ] Remove